src/main/java/com/cloudbees/fm/jenkins/FeatureManagementConfigurationBuilder.java

```diff
@@ -217,7 +219,11 @@ private String getApiToken(String credentialsId) {
             return creds.stream().findFirst().map(c -> c.getSecret().getPlainText()).orElseThrow(() -> new RuntimeException("Could not find credential ID " + credentialsId));
         }
 
-        public ListBoxModel doFillApplicationIdAndNameItems(@QueryParameter String credentialsId) throws IOException {
+        @POST
+        public ListBoxModel doFillApplicationIdAndNameItems(@QueryParameter String credentialsId, @AncestorInPath Item item) throws IOException {
+            if (item != null) {
+                item.checkPermission(Permission.CONFIGURE);
```

Comment:
This can cause trouble with the pipeline snippet generator if used by someone with this permission. Perhaps something like?

```suggestion
                if (!item.hasPermission(Permission.CONFIGURE)) {
                    return new StandardListBoxModel().includeEmptyValue();
                }
```